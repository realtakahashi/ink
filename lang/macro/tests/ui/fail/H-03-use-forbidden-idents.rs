use ink_lang as ink;

#[ink::contract(version = "0.1.0")]
mod forbidden_indents {
    #[ink(storage)]
    struct ForbiddenIndents {}

    impl ForbiddenIndents {
        #[ink(constructor)]
        fn constructor() -> Self {
            Self {}
        }

        #[ink(message)]
        fn message(&self) {
            // All identifiers starting with `__ink` are forbidden to use in ink!.
            let __ink_noop = ();
        }
    }
}

fn main() {}
