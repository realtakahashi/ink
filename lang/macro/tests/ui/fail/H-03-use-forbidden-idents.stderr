error: identifiers starting with `__ink` are forbidden in ink!
  --> $DIR/H-03-use-forbidden-idents.rs:17:17
   |
17 |             let __ink_noop = ();
   |                 ^^^^^^^^^^
